mod-audit-server/src/main/java/org/folio/dao/acquisition/impl/OrderEventsDaoImpl.java

```diff
@@ -44,19 +54,62 @@ public Future<RowSet<Row>> save(OrderAuditEvent orderAuditEvent, String tenantId
     return promise.future();
   }
 
+  @Override
+  public Future<OrderAuditEventCollection> getAcquisitionOrderAuditEventById(String orderId, int limit, int offset, String tenantId) {
+    Promise<RowSet<Row>> promise = Promise.promise();
+    try {
+      String jobTable = formatDBTableName(tenantId, TABLE_NAME);
+      String query = format(GET_BY_ID_SQL, jobTable);
+      Tuple queryParams = Tuple.of(UUID.fromString(orderId), limit, offset);
+      pgClientFactory.createInstance(tenantId).selectRead(query, queryParams, promise);
+    } catch (Exception e) {
+      LOGGER.error("Error getting OrderAuditEvent by id", e);
+      promise.fail(e);
+    }
+    return promise.future().map(rowSet -> rowSet.rowCount() == 0 ? new OrderAuditEventCollection().withTotalItems(0)
+      : mapRowToListOfOrderEvent(rowSet));
+
+  }
+
   private void makeSaveCall(Promise<RowSet<Row>> promise, String query, OrderAuditEvent orderAuditEvent, String tenantId) {
     try {
       pgClientFactory.createInstance(tenantId).execute(query, Tuple.of(orderAuditEvent.getId(),
         orderAuditEvent.getAction(),
         orderAuditEvent.getOrderId(),
         orderAuditEvent.getUserId(),
         orderAuditEvent.getUserName(),
-        orderAuditEvent.getEventDate(),
-        orderAuditEvent.getActionDate(),
-        new JsonObject(orderAuditEvent.getOrderSnapshot())), promise);
+        LocalDateTime.ofInstant(orderAuditEvent.getEventDate().toInstant(),ZoneId.systemDefault()),
+        LocalDateTime.ofInstant(orderAuditEvent.getActionDate().toInstant(),ZoneId.systemDefault()),
```

Comment:
Need to check is it valid date conversion,
basically actionDate exactly should match date that we have in metadata of order's snapshot object

Reply:
fixed.

Reply:
Tested Dates ..
createdDate=2022-12-21T10:00:16.518+00:00

Action date = 2022-12-21 10:00:16.518